### PR TITLE
[bitnami/external-dns] Added extra arguments for Traefik new/legacy resource support

### LINKED ISSUE #25677: [bitnami/external-dns] - Traefik 3.x.x - support resource 'traefik.io' only
### Name and Version

bitnami/external-dns 7.3.0

### What is the problem this feature will solve?

Traefik CRDs `containo.us` are deprecated in favor of `traefik.io`.

[Traefik 3.x.x](https://github.com/traefik/traefik-helm-chart?tab=readme-ov-file#crds-support-of-traefik-proxy) will install `traefik.io` CRDs only.

Enabling `traefik-proxy` as `sources` with Traefik 3.x.x installed from scratch (not upgraded) external-dns fails with error message `failed to sync traefik.containo.us/v1alpha1, Resource=ingressroutes: context deadline exceeded` because CDRs and API group `traefik.containo.us` misses.

In case of Traefik upgrade to version 3.x.x, `containo.us` and `traefik.io` are both present in the cluster, so the issue doesn't occur.

### What is the feature you are proposing to solve the problem?
It seams flag `traefik-disable-legacy` can solve the issue. It there a way to configure it?

### What alternatives have you considered?

Maybe installing CRDs `containo.us` [manually](https://github.com/traefik/traefik-helm-chart?tab=readme-ov-file#upgrade-up-to-27x)

### LINKED ISSUE #27622: failed to sync traefik.containo.us/v1alpha1
### Name and Version

bitnami/external-dns-8.0.0

### What architecture are you using?

arm64

### What steps will reproduce the bug?

This one is not fixed/working:

https://github.com/bitnami/charts/issues/25677

### What do you see instead?
```
time="2024-07-01T19:46:41Z" level=fatal msg="failed to sync traefik.containo.us/v1alpha1, Resource=ingressrouteudps: context deadline exceeded"
```

### PR DESCRIPTION
**Pull request by kamikaze**
### Description of the change

Added additional flag support for new/legacy Traefik resources.
https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/traefik-proxy.md#additional-flags

### Benefits

external-dns will be able to work with both Traefik versions v2 and v3. Also it will be possible to use it with FluxCD which seems doesn't render extraArgs range object well.

### Possible drawbacks

None

### Applicable issues

- fixes #25677
- fixes #27622

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/external-dns/Chart.yaml [hunk 1]
```diff
@@ -28,4 +28,4 @@ maintainers:
 name: external-dns
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/external-dns
-version: 8.0.2
+version: 8.0.3
```
Comment (andresbono) [inline bitnami/external-dns/Chart.yaml:31]:
```suggestion
version: 8.2.0
```

Let's bump the version as minor, as this PR is introducing new parameters in `values.yaml`

Comment (andresbono) [reply bitnami/external-dns/Chart.yaml]: I made the change while resolving merge conflicts

File: bitnami/external-dns/values.yaml [hunk 2]
```diff
@@ -26,6 +26,15 @@ global:
...
       adaptSecurityContext: auto
+    ## Compatibility adaptations for Traefik
+    ##
+    traefik:
+      ## @param global.compatibility.traefik.disableNew Disable listeners on Resources under traefik.io
+      ##
+      disableNew: false
+      ## @param global.compatibility.traefik.disableLegacy Disable listeners on Resources under traefik.containo.us
+      ##
+      disableLegacy: false
```
Comment (andresbono) [inline bitnami/external-dns/values.yaml:37]: I think it is not necessary to define these new values under global, which is a [reserved section in values.yaml](https://helm.sh/docs/chart_template_guide/subcharts_and_globals/#global-chart-values).

What if you set these new values as `.Values.traefik.disableLegacy` and `.Values.traefik.disableNew`?

**Review by andresbono — COMMENTED** [review]
Thank you for your contribution!! 🎉  Could you please check my suggestion?

**Comment by kamikaze** [comment]
> Thank you for your contribution!! 🎉 Could you please check my suggestion?

done

**Review by andresbono — APPROVED** [review]
LGTM. Thank you for addressing the suggestion 👍

**Pull request merged**